### DISCUSSION TITLE
Identifying empty string vs. not given parameter

### DISCUSSION THREAD

**Discussion by jocho-here** [fastapi/fastapi#4357] [Questions] [answered]
Hi,

This is my PATCH endpoint:

```
@router.patch("")
async def create_image_post(
    title: str = Form(...),
    description: Optional[str] = Form(None),
    content_image: Optional[UploadFile] = File(None),
    user: Dict = Depends(get_current_active_user),
):
```

How can FastAPI endpoint accept an empty string and REALLY pass an empty string `""`, not making it as `None`? Currently, when I pass `""` for `description`, it just makes `description` as `None`.

Thanks.

**Comment by floatingpurr** [comment, marked as answer]
Since 

> "Form is a class that inherits directly from Body"

you could define the empty string as the `Form` default value

```python
...
description: Optional[str] = Form(""),
...
```

**Reply by jocho-here** [reply]
In your case, how do I differentiate frontend passing an empty string vs. not passing the description at all? Not even null, but just description not present in the payload.

**Reply by floatingpurr** [reply]
You can explore the incoming [request](https://fastapi.tiangolo.com/advanced/using-request-directly/#using-the-request-directly) directly

**Reply by jocho-here** [reply]
That sounds like a good idea! Thanks!
